testing/esp/esdlcmd/esdlcmd-test.py

```diff
@@ -103,6 +104,108 @@ def validate_results(self):
         logging.debug('TestCaseBase implementation called, no comparison run')
         return False
 
+# This class is the base for the 'transform' commands: ecl and xml.
+# In a future update, investigate refactoring the base classes so there is a single
+# parent with any shared capabilities.
+#
+# When writing to stdout, the key directory should contain a file named 'from-stdout.ecl'
+# that contains the expected output.
+class TestCaseTransformBase:
+    def __init__(self, run_settings, name, command, esdl_file, xsl_path, use_stdout, expected_err=None, options=None):
+        self.run_settings = run_settings
+        self.name = name
+        self.command = command
+        self.esdl_path = (self.run_settings.test_path / 'inputs' / esdl_file)
+        #self.service = service
+        self.xsl_path = xsl_path
+        self.options = options
+        self.output_path = Path(self.run_settings.output_base) / name
+        self.stdout = use_stdout
+        self.expected_err = expected_err
+        if self.stdout:
+            self.args = [
+                str(run_settings.exe_path),
+                self.command,
+                self.esdl_path,
+                '-cde',
+                self.xsl_path,
+            ]
+        else:
+            self.args = [
+                str(run_settings.exe_path),
+                self.command,
+                self.esdl_path,
+                self.output_path,
+                '-cde',
+                self.xsl_path,
+            ]
+
+        if options:
+            self.args.extend(options)
+
+        self.result = None
+
+    def run_test(self):
+        safe_mkdir(self.output_path)
+        logging.debug("Test %s args: %s", self.name, str(self.args))
+        self.result = subprocess.run(self.args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        if self.expected_err != None and self.expected_err == self.result.stderr:
+            success = True
+        elif self.result.returncode != 0:
```

Comment:
Should this check be done first before the expected_err != None check (in Line 153)?

Reply:
Not in this case. The check on 153 captures any tests that are expected to return an error by making sure the error message printed matches what is expected. Then any other test returning an error return code is a failure. I can't make an analogous check of stderr, because the esdl command prints some non-failure messages to stderr.